parallel-install/pkg/deployment/overrideinterceptors.go

```diff
@@ -121,6 +168,75 @@ func (i *DomainNameOverrideInterceptor) findLocalDomain() (domainName string, er
 	return "", nil
 }
 
+// patchCoreDNS takes kubeclient and cluster domain as a parameter to patch coredns config
+// e.g. domainName: `(.*)\.local\.kyma\.dev`
+func patchCoreDNS(kubeClient kubernetes.Interface, domainName string, log logger.Interface) (err error) {
+	coreDNSPatch := CoreDNSPatch{DomainName: domainName}
+	patchTemplate := template.Must(template.New("").Parse(coreDNSPatchTemplate))
+	patchBuffer := new(bytes.Buffer)
+	if err = patchTemplate.Execute(patchBuffer, coreDNSPatch); err != nil {
+		return err
+	}
+
+	// TODO: Refactor
+	err = retry.Do(func() error {
+		_, err := kubeClient.AppsV1().Deployments("kube-system").Get(context.TODO(), "coredns", metav1.GetOptions{})
+		if err != nil {
+			if apierr.IsNotFound(err) {
+				log.Info("CoreDNS not found, skipping CoreDNS config patch")
+				return nil
+			}
+			return err
+		}
+
+		configMaps := kubeClient.CoreV1().ConfigMaps("kube-system")
```

Comment:
how about refactoring as follows (don't pay attention to naming ATM)
```
exists, err := findCoreDNSMap(...)
if err {
  return err
}
if exists {
   err = patchCoreDNSMap(...)
} else {
  err = createCoreDNSMap(...)
}
return err
```

Reply:
Done